src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -337,9 +499,558 @@ private void ReloadLicense()
+                licenseLimit = GenerateLicenseLimit(LimitType.ExternalReplication, details);
+                return false;
+            }
+
+            licenseLimit = null;
+            return true;
+        }
+
+        private static bool IsValidExternalReplication(LicenseStatus licenseStatus, string url)
+        {
+            return licenseStatus.HasExternalReplication || IsValidLocalUrl(url);
+        }
+
+        public bool CanAddRavenEtl(string url, out LicenseLimit licenseLimit)
+        {
+            if (IsValid(out var invalidLicenseLimit) == false)
+            {
+                licenseLimit = invalidLicenseLimit;
+                return false;
+            }
+
+            if (IsValidRavenEtl(_licenseStatus, url) == false)
+            {
+                var details = $"When running {GetLicenseString()}, " +
+                              "only local URLs (which run on localhost, 127.*.*.* or [::1]) can be used for Raven ETL!";
+                licenseLimit = GenerateLicenseLimit(LimitType.RavenEtl, details);
+                return false;
+            }
+
+            licenseLimit = null;
+            return true;
+        }
+
+        private static bool IsValidRavenEtl(LicenseStatus licenseStatus, string url)
+        {
+            return licenseStatus.HasRavenEtl || IsValidLocalUrl(url);
+        }
+
+        public bool CanAddSqlEtl(out LicenseLimit licenseLimit)
+        {
+            if (IsValid(out var invalidLicenseLimit) == false)
+            {
+                licenseLimit = invalidLicenseLimit;
+                return false;
+            }
+
+            if (_licenseStatus.HasSqlEtl == false)
+            {
+                const string details = "The license doesn't allow the usage of SQL ETL!";
+                licenseLimit = GenerateLicenseLimit(LimitType.SqlEtl, details);
+                return false;
+            }
+
+            licenseLimit = null;
+            return true;
+        }
+
+        public bool CanDynamicallyDistributeNodes()
+        {
+            if (IsValid(out var _) == false)
+                return false;
+
+            if (_licenseStatus.HasDynamicNodesDistribution)
+                return true;
+
+            var result = _licenseStatus.Type == LicenseType.None;
+            if (result == false)
+            {
+                const string details = "The license doesn't allow the usage of dynamic nodes distribution!";
+                GenerateLicenseLimit(LimitType.DynamicNodeDistribution, details);
+            }
+
+            return result;
+        }
+
+        public bool CanCreateEncryptedDatabase(out LicenseLimit licenseLimit)
+        {
+            if (IsValid(out var invalidLicenseLimit) == false)
+            {
+                licenseLimit = invalidLicenseLimit;
+                return false;
+            }
+
+            if (_licenseStatus.HasEncryption == false)
+            {
+                const string details = "The license doesn't allow the usage of encryption!";
```

Comment:
Get David to go over all of these texts, we don't want to shout at the user here.

Reply:
done